### DISCUSSION TITLE
Clarifying how to override the default testing directory?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [foundry-rs/foundry#1140] [Q&A] [answered]
In the [Project Layout](https://book.getfoundry.sh/projects/project-layout.html) chapter in the Foundry Book, we are told that:

> The default directory for tests is `src/test/`, where any contract with a function that starts with test is considered to be a test.

Then in the [config README](https://github.com/gakonst/foundry/tree/5916d385dba89cd168f33bee09d5674740045740/config), we are told this:

> The following is a `foundry.toml` file with all configuration options set.
> ```text
> test = 'test'
> ```

Does this mean that Foundry can only look for test contracts directories nested underneath the contracts directory? So if I wanted to load tests from the `foo` directory at the root of my repo, I couldn't do that?

Or that in the latter example, the test directory has been changed from `./src/test` to `./test`?

**Comment by onbjerg** [comment, marked as answer]
No, sorry, this is incorrect. The book is a little out of date.

Our default template places tests in `./src/test` and `foundry.toml` specifies the test directory to be `./test`. Tests in `./src` (and subfolders) are run as well as tests in `./test`.

**Reply by PaulRBerg** [reply]
Gotcha, thanks for confirming!
